src/Tes.Runner/Authentication/CredentialsManager.cs

```diff
@@ -16,7 +17,7 @@ public class CredentialsManager
         private readonly ILogger logger = PipelineLoggerFactory.Create<CredentialsManager>();
 
         private readonly RetryPolicy retryPolicy;
-        private const int MaxRetryCount = 5;
```

Comment:
I'm questioning why there are any retries on creating the token credential object? The only thing in there that throws are arguments that are missing or not well formed. Nothing goes onto the wire (to discover anything actually retriable) until `GetTokenAsync` is called, and that doesn't happen in this class at all.

Reply:
@giventocode